mercure.rst

```diff
@@ -417,39 +432,34 @@ And here is the controller::
     // src/Controller/DiscoverController.php
     namespace App\Controller;
 
-    use Lcobucci\JWT\Configuration;
-    use Lcobucci\JWT\Signer\Hmac\Sha256;
-    use Lcobucci\JWT\Signer\Key;
-    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
-    use Symfony\Component\HttpFoundation\Cookie;
     use Symfony\Component\HttpFoundation\Request;
     use Symfony\Component\HttpFoundation\Response;
-    use Symfony\Component\WebLink\Link;
+    use Symfony\Component\Mercure\Authorization;
+    use Symfony\Component\Mercure\Discovery;
 
-    class DiscoverController extends AbstractController
+    class DiscoverController
     {
+        private $discovery;
+        private $authorization;
+
+        public function __construct(Discovery $discovery, Authorization $authorization)
+        {
+            $this->discovery = $discovery;
+            $this->authorization = $authorization;
+        }
+
         public function __invoke(Request $request): Response
         {
-            $hubUrl = $this->getParameter('mercure.default_hub');
-            $this->addLink($request, new Link('mercure', $hubUrl));
-
-            $key = Key\InMemory::plainText('mercure_secret_key'); // don't forget to set this parameter! Test value: !ChangeMe!
-            $configuration = Configuration::forSymmetricSigner(new Sha256(), $key);
-
-            $token = $configuration->builder()
-                ->withClaim('mercure', ['subscribe' => ["http://example.com/books/1"]]) // can also be a URI template, or *
-                ->getToken($configuration->signer(), $configuration->signingKey())
-                ->toString();
-
-            $response = $this->json(['@id' => '/demo/books/1', 'availability' => 'https://schema.org/InStock']);
-            $cookie = Cookie::create('mercureAuthorization')
-                ->withValue($token)
-                ->withPath('/.well-known/mercure')
-                ->withSecure(true)
-                ->withHttpOnly(true)
-                ->withSameSite('strict')
-            ;
-            $response->headers->setCookie($cookie);
+            $this->discovery->addLink($request);
+
+            $response = new JsonResponse([
```

Comment:
I suggest using the `$this->json()` method provided by `AbstractController` for consistency too.